consensus/src/marshal/mod.rs

```diff
@@ -76,6 +76,18 @@ pub trait SchemeProvider: Clone + Send + Sync + 'static {
     fn scheme(&self, epoch: Epoch) -> Option<Arc<Self::Scheme>>;
 }
 
+/// An update reported to the application, either a new tip or a finalized block.
+///
+/// Tips are reported as soon as known, whether or not we hold all blocks to that height.
+/// Finalized blocks are reported to the application in monotonically increasing order.
+#[derive(Clone, Debug)]
```

Comment:
At first, I thought this might just be a "lazy" way to just not listen to the consensus reporter but then realized that providing height here (along with commitment) is actually introducing new functionality.

Reply:
The downside is that we need a block (rather than just a certificate) to produce this, but in the happy path we should almost always have this as soon as we have the cert.